nexus/db-queries/src/db/datastore/oximeter.rs

```diff
@@ -15,16 +15,32 @@ use crate::db::model::OximeterInfo;
 use crate::db::model::ProducerEndpoint;
 use crate::db::pagination::paginated;
 use crate::db::pagination::Paginator;
+use crate::db::queries;
 use async_bb8_diesel::AsyncRunQueryDsl;
 use chrono::DateTime;
 use chrono::Utc;
 use diesel::prelude::*;
+use diesel::result::DatabaseErrorKind;
+use diesel::result::Error as DieselError;
+use diesel::sql_types;
+use nexus_db_model::ProducerKindEnum;
 use omicron_common::api::external::DataPageParams;
 use omicron_common::api::external::Error;
 use omicron_common::api::external::ListResultVec;
 use omicron_common::api::external::ResourceType;
 use uuid::Uuid;
 
+/// Type returned when reassigning producers from a (presumably defunct)
```

Comment:
This seems to hedge about the state of the collector, which I also see in other places. Is there a reason we don't know the state more concretely? I'm uncertain what to take away from the "presumably defunct" bit.

Reply:
The hedging is because `oximeter_reassign_all_producers()` doesn't actually require that the collector being reassigned from is expunged/deleted; it could still be live! That would be weird, because since reassignment randomly chooses from available collectors, it would get some of its own producers back again.

A couple options:

* Just remove the parenthetical entirely and leave the code as-is
* Modify the reassignment query to require that the collector be expunged

I'm inclined to go with the first one; in hindsight I think the hedging is technically correct but not at all useful.

Reply:
Yeah, option 1 sounds good to me too.